src/libexpr/primops/fetchTree.cc

```diff
@@ -182,45 +182,208 @@ static RegisterPrimOp primop_fetchTree({
     .name = "fetchTree",
     .args = {"input"},
     .doc = R"(
-      Fetch a source tree or a plain file using one of the supported backends.
-      *input* must be a [flake reference](@docroot@/command-ref/new-cli/nix3-flake.md#flake-references), either in attribute set representation or in the URL-like syntax.
-      The input should be "locked", that is, it should contain a commit hash or content hash unless impure evaluation (`--impure`) is enabled.
+      Fetch a file system tree or a plain file using one of the supported backends and return a [store path](@docroot@/glossary.md#gloss-store-path) with metadata.
```

Comment:
Not just the store path, but an attribute set with `outPath` as the store path and `narHash` as well.